src/cargo/util/toml/manifest_cache.rs

```diff
@@ -0,0 +1,102 @@
+use std::collections::{
```

Comment:
maybe this can just be in `toml/mod.rs`. Me-two-weeks-ago though this deserved its own file, but I'm not sure anymore.

Reply:
Nah `mod.rs` is pretty huge as-is, so I think it's fine for this to be its own file.